### PR TITLE
Added support for switch and break statements

### PR DESCRIPTION
**Pull request by thosakwe**
Resolves https://github.com/dart-lang/code_builder/issues/48. 😄

### CODE REVIEW

**Comment by thosakwe** [comment]
Passing on stable, failing on dev. The reasons it's failing on dev don't seem to be related to my commits

File: lib/src/builders/statement/break.dart [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import 'package:analyzer/analyzer.dart';
```
Comment (matanlurey) [inline lib/src/builders/statement/break.dart]: Please add the BSD header.

File: lib/src/builders/statement/break.dart [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import 'package:analyzer/dart/ast/standard_ast_factory.dart';
+import 'package:code_builder/src/builders/shared.dart';
+import 'package:code_builder/src/builders/statement.dart';
+import 'package:code_builder/src/tokens.dart';
+
+/// Represents a break statement AST.
+abstract class BreakStatementBuilder implements StatementBuilder {
```
Comment (matanlurey) [inline lib/src/builders/statement/break.dart]: Can we just replace this with a singleton instance, i.e.

const breakStatement = const _BreakStatementBuilder();

Comment (thosakwe) [inline lib/src/builders/statement/break.dart]: Sure, I'll apply the edits when I get home today.

Comment (thosakwe) [inline lib/src/builders/statement/break.dart]: Ok, @matanlurey ,  I have made the edits.

**Comment by matanlurey** [comment]
Thanks for the contribution. Dev-channel failure looks related to a pkg/test regression.

**Comment by thosakwe** [comment]
Of course, any time! :)

**Pull request merged**